### DISCUSSION TITLE
How to share authenticated user information to separate express backend server

### DISCUSSION THREAD

**Discussion by myguydude** [nextauthjs/next-auth#4914] [Help]
I currently have set up my nextjs application with `next-auth`, using 2 providers: Google & Facebook

I have my **nextjs application** running on : `domain.com`
I don't want to use nextjs api routes, so I have a separate **express server** running on : `api.domain.com`

On the front-end, when I send an api request to my `api.domain.com` backend, I attach the `__Secure-next-auth.session-token` or `next-auth.session-token` (depending on the envrionment) to the every request I send.

When I receive a request on my `api.domain.com`, I check for either of those cookies and I decode it like this:

```
 const jwt = await decode({
      token: tokenCookie,
      secret: process.env.JWT_SECRET,
    })

const userID = jwt.userID

-----------

export async function decode(params: JWTDecodeParams): Promise<JWT | null> {
  const { token, secret } = params
  if (!token) return null
  const encryptionSecret = await getDerivedEncryptionKey(secret)
  const { payload } = await jwtDecrypt(token, encryptionSecret, {
    clockTolerance: 15,
  })
  return payload
}

async function getDerivedEncryptionKey(secret: string | Buffer) {
  return await hkdf(
    'sha256',
    secret,
    '',
    'NextAuth.js Generated Encryption Key',
    32,
  )
}
```

(This code is based on) : https://github.com/nextauthjs/next-auth/blob/0bf955a63d3a2571530399027fbb7f023adcb2f8/packages/next-auth/src/jwt/index.ts#L29

Doing this, I can successfully retrieve the **userID stored in the jwt token** and carry on on my custom backend server to authorize an action or not.

Is there a better way to **share the authenticated user on my nextjs application to my separate express backend server**? 
The way I do it currently, works perfectly, but it feels kinda hacky.

**Comment by antonmihaylov** [comment]
I use a similar pattern, except that It directly calls getToken from next-auth and everything is wrapped as a custom passport strategy

```ts
import { getToken } from 'next-auth/jwt'


// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-expect-error
globalThis.Headers = class Headers {}


export const authStrategy = new CustomStrategy((req, callback) => {
  void (async () => {
    try {
      const token = await getToken({
        secret: config.jwt.secret,
        req,
        logger: nextAuthLogger,
        cookieName: config.jwt.sessionCookieName,
        secureCookie: config.isProd,
      })

      callback(null, token)
    } catch (error) {
      callback(error, false)
    }
  })()
})
```

**Comment by mattbarnicle** [comment]
i'm also considering implementing something like this if i end up using NextAuth for my project. i've seen a simlar question asked several times in the discussions, and it seems like there must be a decent number of people who need this type of functionality. it would be nice to get some feedback from some developers of the project to weigh in on this, or ideally to be mentioned in the docs somewhere. appreciate it if someone can do that.
